#!/usr/bin/env python
PACKAGE = "siar_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("T",    double_t,    0, "Controller period", 0.05, 0.01,   0.2)
gen.add("delta_T",  double_t,    0, "Simulation timestep",    0.05, 0.005, 0.2)
gen.add("T_hor",  double_t,    0, "Simulation horizon",    2.0, 0.1,   5)
gen.add("operation_mode",  int_t,         0, "Operation mode: 0-MANUAL 1-CENTER", 0, 0, 10)
gen.add("v_max", double_t, 0, "Maximum speed of the robot (m/seg)", 0.5, 0.1, 1.5)
gen.add("alpha_max", double_t, 0, "Maximum angular speed of the robot (rad/seg)", 0.5, 0.1, 2.5)
gen.add("d_axis", double_t, 0, "Axis distance of the robot (m)", 0.56, 0.1, 2.5)
gen.add("w_dist", double_t, 0, "Distance weight", 1.0, 0.1, 1000)
gen.add("w_safe", double_t, 0, "Safe weight", 10.0, 0.1, 1000)

exit(gen.generate(PACKAGE, "siar_controller", "SiarController"))

